fsharp-backend/src/LibExecutionStdLib/LibList.fs

```diff
@@ -1490,14 +1492,18 @@ let fns : List<BuiltInFn> =
                 | DBool b -> return b
                 | (DIncomplete _
                 | DErrorRail _
-                | DError _) as dv -> return false
+                | DError _) as dv ->
```

Comment:
I'm having a few conflicting thoughts:

1. The mechanism for returning things should be the same in both cases (fakeval and type error)
2. It should return early upon finding a faekval
3. Do we want to keep using Exceptions in stdlib code?

Reply:
1. I'm not sure what you mean here - what are the 2 'mechanisms'? Exceptions and dvals?
2. Resolved - decided to not extract out `partition` in `Prelude`
3. I think it's OK in cases like these where we see those exceptions being preventable (i.e. with type checking before we end up here)

Reply:
> I'm not sure what you mean here - what are the 2 'mechanisms'? Exceptions and dvals?

Yeah, if there's a new type error than we throw. If there's an existing type error we return it. I think I'm maybe being picky here - we probably do it like this in many other places.